app/models.py

```diff
@@ -0,0 +1,106 @@
+from flask_sqlalchemy import SQLAlchemy
+from flask_migrate import Migrate
+from sqlalchemy.ext.declarative import declared_attr
+import sqlalchemy as sa
+
+migrate = Migrate()
+
+
+class IdModel(object):
+    @declared_attr
+    def id(cls):
+        for base in cls.__mro__[1:-1]:
+            if getattr(base, "__table__", None) is not None:
+                type = sa.ForeignKey(base.id)
+                break
+        else:
+            type = sa.Integer
+
+        return sa.Column(type, primary_key=True)
+
+
+db = SQLAlchemy(model_class=IdModel)
+
+
+class NameDescriptionMixin(object):
+    name = db.Column(db.String(128))
+    description = db.Column(db.Text())
+
+
+class SkillsMixin(object):
+    @declared_attr
+    def skill_id(cls):
+        return db.Column(db.ForeignKey("skill.id"))
+
+    @declared_attr
+    def skill_level_id(cls):
+        return db.Column(db.ForeignKey("skill_level.id"))
+
+
+class Skill(db.Model, NameDescriptionMixin):
+    core = db.Column(db.Boolean())
+    profession_id = db.Column(db.ForeignKey("profession.id"))
+
+
+class SkillLevel(db.Model):
+    skill_id = db.Column(db.ForeignKey("skill.id"))
+    assessment_id = db.Column(db.ForeignKey("assessment.id"))
+    value = db.Column(db.Integer())
+
+
+class Profession(db.Model, NameDescriptionMixin):
+    pass
+
+
+class Role(db.Model):
+    title = db.Column(db.String(128))
+
+    profession_id = db.Column(db.ForeignKey("profession.id"))
+
+
+class Posting(db.Model):
+    candidate_id = db.Column(db.ForeignKey("candidate.id"))
+    role_id = db.Column(db.ForeignKey("role.id"))
+
+    date_started = db.Column(db.Date())
+    date_ended = db.Column(db.Date())
+    rotation_number = db.Column(db.Integer())
+
+
+class Assessment(db.Model):
+    posting_id = db.Column(db.ForeignKey("posting.id"))
+
+    date = db.Column(db.Date())
+    overall_mark = db.Column(db.String(56))
+
+    posting = db.relationship("Posting")
+
+    def candidate(self):
+        return Candidate.query.get(self.posting.candidate_id)
+
+
+class Candidate(db.Model):
+    first_name = db.Column(db.String(128))
+    last_name = db.Column(db.String(128))
+    email_address = db.Column(db.String(120), unique=True)
+    secondary_email_address = db.Column(db.String(120), unique=True)
+    joining_date = db.Column(db.Date())
+    completed_fast_stream = db.Column(db.Boolean())
```

Comment:
You don't seem to refer overtly to the fast stream anywhere else e.g. calling the repo Graduate Rotator not 'Fast Stream rotator. 

If you want the app to be a blueprint which can be used for other graduate schemes, then maybe calling the field `completed_graduate_scheme` or `completed_scheme` would be more appropriate? 

If not, maybe calling the app 'Fast stream rotator' and being more explicit that it's only for the fast stream would make it clearer?

Reply:
Good spot - I'll drop this column